cmd/clusterctl/cmd/generate_yaml.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+)
+
+type generateYAMLOptions struct {
+	url           string
+	listVariables bool
+}
+
+var gyOpts = &generateYAMLOptions{}
+
+var generateYamlCmd = &cobra.Command{
```

Comment:
I noticed that not all the clusterctl commands are documented in the book. Do we want to document this command?

Reply:
Yes, that'd be great to comment here as well

Reply:
I'll add a section in our documentation. However, I think a comment here is unnecessary since we have the description of the command below.